### PR TITLE
fix excessive nesting in MQClientInstance.java

### PR DESCRIPTION
**Pull request by SeaItFover**
**Make sure set the target branch to `develop`**

## What is the purpose of the change

[code style] :fix excessive nesting 

## Brief changelog

fix excessive nesting 

## Verifying this change

fix excessive nesting 

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [x] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

File: client/src/main/java/org/apache/rocketmq/client/impl/factory/MQClientInstance.java [hunk 1]
```diff
@@ -561,42 +561,43 @@ private void sendHeartbeatToAllBroker() {
...
+            for (Entry<Long, String> singleBrokerInstance : oneTable.entrySet()) {
+                Long id = singleBrokerInstance.getKey();
+                String addr = singleBrokerInstance.getValue();
+                if (addr == null) {
+                    continue;
+                }
+                if (consumerEmpty) {
+                    if (id != MixAll.MASTER_ID)
+                        continue;
+                }
```
Comment (aaron-ai) [inline client/src/main/java/org/apache/rocketmq/client/impl/factory/MQClientInstance.java:583]:
```suggestion
                if (consumerEmpty && MixAll.MASTER_ID != id) {
                    continue;
                }
```

**Review by aaron-ai — APPROVED** [review]

File: pom.xml [hunk 2]
```diff
@@ -152,12 +152,13 @@
...
         <jacoco-maven-plugin.version>0.8.5</jacoco-maven-plugin.version>
-        <maven-surefire-plugin.version>2.19.1</maven-surefire-plugin.version>
+        <maven-surefire-plugin.version>2.22.1</maven-surefire-plugin.version>
+        <findbugs-maven-plugin.version>3.0.4</findbugs-maven-plugin.version>
         <sonar-maven-plugin.version>3.0.2</sonar-maven-plugin.version>
         <spotbugs-plugin.version>4.2.2</spotbugs-plugin.version>
         <maven-assembly-plugin.version>3.0.0</maven-assembly-plugin.version>
         <maven-javadoc-plugin.version>2.10.4</maven-javadoc-plugin.version>
-        <maven-failsafe-plugin.version>2.19.1</maven-failsafe-plugin.version>
+        <maven-failsafe-plugin.version>2.22.1</maven-failsafe-plugin.version>
```
[the remark this answers is not in the document]
Comment (SeaItFover) [reply pom.xml:161]: for the previous version, srvutil module test case cant pass, this is the version bug (stackoverflow result).

[the remark this answers is not in the document]
Comment (SeaItFover) [reply pom.xml:161]: Because i just modify the code style, why the test case cant pass..

File: pom.xml [hunk 3]
```diff
@@ -152,12 +152,13 @@
         <maven-resources-plugin.version>3.0.2</maven-resources-plugin.version>
         <coveralls-maven-plugin.version>4.3.0</coveralls-maven-plugin.version>
         <jacoco-maven-plugin.version>0.8.5</jacoco-maven-plugin.version>
-        <maven-surefire-plugin.version>2.19.1</maven-surefire-plugin.version>
+        <maven-surefire-plugin.version>2.22.1</maven-surefire-plugin.version>
+        <findbugs-maven-plugin.version>3.0.4</findbugs-maven-plugin.version>
         <sonar-maven-plugin.version>3.0.2</sonar-maven-plugin.version>
```
Comment (ni-ze) [inline pom.xml:157]: Sorry, I do not get why modify here?

Comment (SeaItFover) [reply pom.xml]: I have revert the commit.

**Review by aaron-ai — APPROVED** [review]

**Pull request merged**